server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v671/DeleteHistoryOfTestFileMeasures.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.platform.db.migration.version.v671;
+
+import java.sql.SQLException;
+import org.sonar.db.Database;
+import org.sonar.db.dialect.H2;
+import org.sonar.db.dialect.MySql;
+import org.sonar.server.platform.db.migration.step.DataChange;
+
+public class DeleteHistoryOfTestFileMeasures extends DataChange {
+
+  private static final String MYSQL_SQL = "delete pm from project_measures pm " +
+    "inner join snapshots s on s.uuid = pm.analysis_uuid " +
+    "inner join projects c on c.uuid = pm.component_uuid " +
+    "where s.islast=? and c.qualifier in ('UTS', 'FIL')";
+
+  private static final String OTHER_SQL = "delete from project_measures " +
+    "where exists ( " +
+    "  select 1 from project_measures pm " +
+    "  inner join snapshots s on s.uuid = pm.analysis_uuid " +
+    "  inner join projects c on c.uuid = pm.component_uuid " +
+    "  where s.islast=? and c.qualifier in ('UTS', 'FIL')" +
+    ")";
+
+  private static final String H2_SQL = "delete from project_measures " +
+    "where id in ( " +
+    "  select pm.id from project_measures pm " +
+    "  inner join snapshots s on s.uuid = pm.analysis_uuid " +
+    "  inner join projects c on c.uuid = pm.component_uuid " +
+    "  where s.islast=? and c.qualifier in ('UTS', 'FIL') " +
```

Comment:
(occurance 2) 💣   Add a `where` clause, referring to the measure to delete. Currently the inner select is not correlated and this will **delete all unit test and file measures**!